Automattic/jetpack
# mirrors: Add the ability to auto-publish to WordPress.org SVN

#### Changes proposed in this Pull Request:

As requested in #21558, this adds the ability to have a plugin
automatically publish tags to WordPress.org SVN.

It doesn't enable it for any plugins. That's left for a later PR.

#### Jetpack product discussion


#21558 

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* You can run the script manually after setting the appropriate environment variables, something like this:
  * Create a temporary directory, and cd to it.
  * `git clone --depth=1 git@github.com:Automattic/jetpack-production src`
  * Set the appropriate environment variables:
    - `export GITHUB_REF=refs/tags/v10.3` or whatever tag you're wanting to test.
    - `export WPSVN_USERNAME=...`
    - `export WPSVN_PASSWORD=...`
    - If you want it to actually push to SVN, do `export CI=1`. Otherwise `export -n CI` to make sure it's _not_ set.
  * Run the `.github/files/gh-wp-svn-autopublish/files/wp-svn-autopublish.sh` script. Check that it does the right thing.
